libkbfs/tlf_handle_extension.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/keybase/go-codec/codec"
+)
+
+const (
+	// TlfHandleExtensionDateFormat is the date format for the TlfHandleExtension string.
+	TlfHandleExtensionDateFormat = "2006-01-02"
+	// TlfHandleExtensionDateRegex is the regular expression matching the TlfHandleExtension
+	// date member in string form.
+	TlfHandleExtensionDateRegex = "2[0-9]{3}-[0-9]{2}-[0-9]{2}"
+	// TlfHandleExtensionNumberRegex is the regular expression matching the TlfHandleExtension
+	// number member in string form.
+	TlfHandleExtensionNumberRegex = "[0-9]+"
+	// TlfHandleExtensionTypeRegex is the regular expression matching the TlfHandleExtension
+	// type string.
+	TlfHandleExtensionTypeRegex = "conflicted copy|finalized"
+	// TlfHandleExtensionFormat is the formate string for a TlfHandleExtension.
+	TlfHandleExtensionFormat = "(%s %s%s)"
+)
+
+// TlfHandleExtensionType is the type of extension.
+type TlfHandleExtensionType int
+
+const (
+	// TlfHandleExtensionConflict means the handle conflicted as a result of a social
+	// assertion resolution.
+	TlfHandleExtensionConflict TlfHandleExtensionType = iota
+	// TlfHandleExtensionFinalized means the folder ended up with no more valid writers as
+	// a result of an account reset.
+	TlfHandleExtensionFinalized
+	// TlfHandleExtensionUnknown means the type is unknown.
+	TlfHandleExtensionUnknown
+)
+
+// String implements the fmt.Stringer interface for TlfHandleExtensionType
+func (et TlfHandleExtensionType) String() string {
+	switch et {
+	case TlfHandleExtensionConflict:
+		return "conflicted copy"
```

Comment:
I think "conflicted copy" and "finalized" should both be constants, since you use them in two places.  You could construct the regex from them as well.